digits/model/tasks/caffe_train.py

```diff
@@ -39,6 +39,50 @@
 CAFFE_DEPLOY_FILE = 'deploy.prototxt'
 
 @subclass
+class DigitsTransformer(caffe.io.Transformer):
```

Comment:
Inference is running in a dedicated process therefore `resize` always has a fixed value, for a given run. The mean image is only resized if it has a different shape from the image to perform inference on. So if you are doing inference on a batch of images of the same size, mean image resizing should only occur once. On the next inference job everything gets reset so hopefully the issue you described can't occur.

Note: doing inference on a batch of images that have variable size will cause problems. Perhaps this should be listed as a known limitation - supporting this is very tricky.